src/aiokatcp/sensor.py

```diff
@@ -214,7 +218,18 @@ def set_value(
         timestamp
             The time at which the sensor value was determined (seconds).
             If not given, it defaults to :func:`time.time`.
+
+        Raises
+        ------
+        TypeError
+            If the incoming `value` type is not compatible with the sensor's
+            core type.
         """
+        if not issubclass(type(value), self._core_type):
```

Comment:
This is a shorter version of the same thing:
```suggestion
        if not isinstance(value, self._core_type):
```

Reply:
Touché, Bruce! This was languishing in my review while I looked for examples in SDP code :joy: